packages/unpublished-test-infra/addon-test-support/assert-all-deprecations.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-console */
+import { DEBUG } from '@glimmer/env';
+import QUnit from 'qunit';
+import config from 'ember-get-config';
+
+const { ASSERT_ALL_DEPRECATIONS } = config;
+
+const ALL_ASSERTED_DEPRECATIONS = [];
+
+export default function configureAssertAllDeprecations() {
+  QUnit.begin(() => {
+    function assertAllDeprecations(assert) {
+      if (typeof assert.test.expected === 'number') {
+        assert.test.expected += 1;
+      }
+      assert.expectNoDeprecation(undefined, undefined, deprecation => {
+        // only assert EmberData deprecations
+        const id = deprecation.options.id.toLowerCase();
+        const isEmberDataDeprecation =
+          id.includes('ds.') ||
+          id.includes('emberdata') ||
+          id.includes('ember-data') ||
+          id.includes('mismatched-inverse-relationship-data-from-payload');
+
+        if (!ASSERT_ALL_DEPRECATIONS && !isEmberDataDeprecation) {
+          console.warn('Detected Non-Ember-Data Deprecation:', deprecation.message, deprecation.options.stacktrace);
+        }
+        if (ASSERT_ALL_DEPRECATIONS) {
+          ALL_ASSERTED_DEPRECATIONS.push((deprecation.options && deprecation.options.id) || deprecation);
```

Comment:
we don't know from within this filter method whether the deprecation is handled, only that it is encountered.

I think we can just ignore things that match `isEmberDataDeprecation` since we already assert all our own in regular tests and thus can be sure they are handled.  This will reduce the noise in the report.

Reply:
Ok!